[Federation]Fix panic: index out of range

FengyunPan opened the pull request:
When the number of clusterStatusNew's Conditions is different from
the number of clusterStatusOld's Conditions, clustercontroller
will panic. We should check it before comparing.

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/44526)
<!-- Reviewable:end -->

FengyunPan commented:
/release-note-none

Commit:
[Federation]Fix panic: index out of range

When the number of clusterStatusNew's Conditions is different from
the number of clusterStatusOld's Conditions, clustercontroller
will panic. We should check it before comparing.

quinton-hoole commented:
@k8s-bot ok to test

quinton-hoole commented:
/lgtm

@FengyunPan thanks very much for all of your very useful PR's!
